wallet/src/Dscp/Wallet/Backend.hs

```diff
@@ -29,14 +31,17 @@ createWalletFace serverAddress sendEvent = buildComponent_ "Wallet" $ do
         , walletListKeys = listKeys
         , walletSendTx = sendTx wc sendEvent
         , walletTxFee = unFees . estimateFees feeConfig . toList
-        , walletGetBalance = getBalance wc
-        , walletGetTxHistory = getTxHistory wc
+        , walletGetBalance = getBalance wc sendEvent
+        , walletGetTxHistory = getTxHistory wc sendEvent
         }
 
-sendStateUpdateEvent :: (WalletEvent -> IO ()) -> IO ()
+sendStateUpdateEvent :: SendEvent -> IO ()
```

Comment:
If you start accepting callbacks in lots of places, it means that you probably should think of wrapping them in a monad, or the code will get ugly quickly.
Not critical, of course.